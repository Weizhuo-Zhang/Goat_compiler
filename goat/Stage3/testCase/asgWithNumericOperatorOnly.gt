
# Testing Goat assignments

proc main()
    int a;
    bool b;
begin
    a := 42;
    write a;
    write "\n";
    a := a;
    write a;
    write "\n";
    a := (4*5*6) + (1*2*3);
    write a;
    write "\n";
    a := (4/5/6) - (1/2/3);
    write a;
    write "\n";
    a := (4+5+6) * (1+2+3);
    write a;
    write "\n";
    a := (4-5-6) / (1-2-3);
    write a;
    write "\n";
    a := -0;
    write a;
    write "\n";
    a := -(1*2*3);
    write a;
    write "\n";
end
