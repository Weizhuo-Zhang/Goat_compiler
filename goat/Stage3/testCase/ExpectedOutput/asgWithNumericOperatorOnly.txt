42
42
126
0
90
1
